### PR TITLE
doc/dev: define what mempools we use in bluestore

### PR DESCRIPTION
**Pull request by rzarzynski**
Signed-off-by: Deepika <dupadhya@redhat.com>

----
This commit has been dissected from https://github.com/ceph/ceph/pull/40066 to unblock the documentation-related parts of it.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by ideepika — APPROVED** [review]

File: doc/dev/mempool_accounting.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  allow RocksDB to store its “files” and share the same raw device(s) with
+  BlueStore.
+- buffer_anon: stores arbitrary buffer data.
+- buffer_meta: all the metadata associated with buffer anon buffers.
+- bluestore_cache_data: mempool for writing and writing deferred.
+- bluestore_cache_onode: meta information of object node stored directed in
+  bluestore cache.
+- bluestore_cache_meta: key under PREFIX_OBJ where we are stored.
+- bluestore_cache_other: right now account's for
+
```
Comment (anthonyeleven) [inline doc/dev/mempool_accounting.rst:32]: I think the apostrophe is superfluous and the tail end of the description is missing

**Comment by anthonyeleven** [comment]
Really nice to see these described.  Formatting suggestion:
* `backtick` the pool names
*  Capitalize blue store->BlueStore when it isn’t part of a type name.

File: doc/dev/mempool_accounting.rst (hunk 1 above)
Comment (ideepika) [reply doc/dev/mempool_accounting.rst:32]: hey @anthonyeleven this was intentional, as following being sub-points for bluestore_cache_other:
image

**Comment by ideepika** [comment]
@rzarzynski @anthonyeleven  amended PR addressing comment feel free to squash/comment here

**Comment by rzarzynski** [comment]
Will squash today. Thanks, @ideepika!

File: doc/dev/mempool_accounting.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+A memory pool is a method for accounting the consumption of memory of
+a set of containers.
+Each memory pool tracks the number of bytes and items it contains.
+Memory pools are statically declared and serves the purpose of identifying
+memory related checks in BlueStore.
+
+Some common mempools that we can track
+--------------------------------------
+
+- bloom_filter: Ceph uses bloom filters to account for tracking objects already
```
Comment (anthonyeleven) [inline doc/dev/mempool_accounting.rst:12]: Suggest
    ``bloom_filter``: Ceph uses ...
and similarly for other list items to highlight that these are proper names.

Comment (rzarzynski) [reply doc/dev/mempool_accounting.rst]: Formatted! Please take another look.

File: doc/dev/mempool_accounting.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
+What is a mempool?
+------------------
+A memory pool is a method for accounting the consumption of memory of
+a set of containers.
```
Comment (anthonyeleven) [inline doc/dev/mempool_accounting.rst:4]: We have mempools without containers too, suggest rewording to "Memory pools (mempools) show the details of memory consumption by each Ceph daemon".

Comment (ideepika) [reply doc/dev/mempool_accounting.rst:4]: @anthonyeleven good that we can assess what idea it conveys, the actual meaning here is memory accounting of C++ classes and containers used in mostly bluestore component. Perhaps,

```suggestion
Memory pools (mempools) show the details of memory consumption of C++ classes and containers, which can then be used to assess memory leaks, and other insights around memory usage with low overhead.
```

**Comment by anthonyeleven** [comment]
Hoookay sorry for the delay.  I foolishly hit the accept button on your proposed change, which created a new commit, which wasn't signed, so the check failed which I couldn't override.  I managed to amend that commit; when this round of checks completes I'll merge it.  If there are additional refinements please open a new PR after the merge, as experience shows that in a situation like this, it's best to merge while we're ahead as the local and remote states can become ... tangled.
